.github/workflows/main-pull.yml

```diff
@@ -114,4 +90,48 @@ jobs:
     #   run: cabal haddock all
 
     # - name: Prepare package for publishing
-    #   run: cabal sdist all
\ No newline at end of file
+    #   run: cabal sdist all
+  stylish-haskell-job:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        cabal: [3.14.1.1]
+        ghc: ["9.10"]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4.1.1
+
+    - name: Set up Haskell
+      uses: haskell-actions/setup@v2.6.2
+      id: setup
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: ${{ matrix.cabal }}
+        cabal-update: true
+    
+    - name: Install stylish-haskell
+      run: |
+        cabal install stylish-haskell
+
+    - name: Cargo clippy
```

Comment:
Don't know if this should go in the "stylish-haskell-job" or if the job itself should he named differently, like "haskell-rust-lint"